service.php

```diff
@@ -263,6 +259,20 @@ function get_credentials() {
 		return apply_filters( 'keyring_service_credentials', $creds, $this->get_name() );
 	}
 
+		/**
+	 * Set the basic credential properties based
+	 * on an Array of credentials
+	 *
+	 * @param Mixed $credentials
+	 */
+	function init_credentials( $credentials ) {
+		if ( ! empty( $credentials ) && is_array( $credentials ) ) {
+			$this->app_id = $credentials['app_id'] ? $credentials['app_id'] : null;
```

Comment:
Should these use `isset`? I'm not sure, but we might get some warnings otherwise.

Reply:
Yes, caught that and changed in 81ae051b0ff362ba9ccc2e7c0cc33377b4592964 as well

Reply:
Weird, I'm not seeing that for some reason. I was thinking we should change these checks to something like:

```
$this->app_id = isset( $credentials['app_id'] ) ? $credentials['app_id'] : null;
```

Although again we might want to use `empty` in order to catch other falsey values.

```
$this->app_id = ! empty( $credentials['app_id'] ) ? $credentials['app_id'] : null;
```

Reply:
`array_key_exists` is the most appropriate here tbh. You can't know if falsey values are inappropriate, and `array_key_exists` saves you from making this decision.

Reply:
Changed to `empty` check here: 429a3280ce5ce743aeb756d6c1445e314cdf3159
`array_key_exists` could have been an option too but this should be fine for our needs.